File: Tools/Lua客户端C#版/Lua/ScutReader.lua
ScutReader ={}
function ScutReader:new()
	local instance = {}
	setmetatable(instance, self)
	self.__index = self
	
end
function ScutReader:readAction()
    return ScutReader_readAction()
end

function ScutReader:getInt()
    return ScutReader_getInt()
end
function ScutReader:getWORD()
    return ScutReader_getWORD()
end
function ScutReader:getByte()
    return ScutReader_getByte()
end
function ScutReader:readString()
    return ScutReader_readString()
end

function ScutReader:recordBegin()
    return ScutReader_recordBegin()
end

function ScutReader:recordEnd()
    return ScutReader_recordEnd()
end

function ScutReader:readErrorMsg()
    return ScutReader_readErrorMsg()
end

function ScutReader:readErrorCode()
    return ScutReader_readErrorCode()
end


function ScutReader:getResult()
    return ScutReader_getResult()
end

File: Source/Tools/ScutClient/ScutClientSimulator/Net/ScutReader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ZyGames.Framework.RPC.IO;

namespace Scut.Client.Net
{
    public class ScutReader
    {
        private static ScutReader instance = new ScutReader();

        public static ScutReader GetIntance()
        {
            return instance;
        }

        private MessageStructure _dataReader;
        private MessageHead _head;

        private ScutReader()
        {
        }

        public MessageStructure Reader
        {
            get { return _dataReader; }
            set { _dataReader = value; }
        }

        #region read method

        public bool getResult()
        {
            if (Reader.Offset == 0)
            {
                _head = instance.Reader.ReadHeadGzip();
            }
            return !_head.Faild;
        }

        public int readAction()
        {
            return _head.Action;
        }

        public int readErrorCode()
        {
            return _head.ErrorCode;
        }

        public string readErrorMsg()
        {
            return _head.ErrorInfo;
        }

        public int getWORD()
        {
            return Reader.ReadShort();
        }

        public int getInt()
        {
            return Reader.ReadInt();
        }

        public int getByte()
        {
            return Reader.ReadByte();
        }

        public long getLong()
        {
            return Reader.ReadLong();
        }

        public double getDouble()
        {
            return Reader.ReadDouble();
        }

        public string readString()
        {
            return Reader.ReadString();
        }

        public void recordBegin()
        {
            Reader.RecordStart();
        }

        public void recordEnd()
        {
            Reader.RecordEnd();
        }

        #endregion

    }
}
